File: app/src/main/java/com/letianpai/robot/wificonnet/callback/KeyPressCallback.java
package com.letianpai.robot.wificonnet.callback;

import java.util.ArrayList;

/**
 * 按键功能回调
 * @author liujunbin
 */
public class KeyPressCallback {
    public static final int KEY_TYPE_FUNCTION = 10010;
    public static final int KEY_TYPE_VALUE = 10011;

    private ArrayList<KeyPressCallbackListener> keyPressCallbackListenerList = new ArrayList<>();

    private static class KeyPressCallbackHolder {
        private static KeyPressCallback instance = new KeyPressCallback();
    }

    public static KeyPressCallback getInstance() {
        return KeyPressCallbackHolder.instance;
    }

    private KeyPressCallback() {

    }

    public void setKeyPressCallbackListener(KeyPressCallbackListener keyPressCallbackListener) {
        this.keyPressCallbackListenerList.add(keyPressCallbackListener);
    }

    public interface KeyPressCallbackListener {
        void onKeyPressed(int keyType,String keyValue);
    }

    public void pressKey(int keyType,String keyValue) {
        for (int i = 0; i < keyPressCallbackListenerList.size();i++){
            if (keyPressCallbackListenerList.get(i) != null) {
                keyPressCallbackListenerList.get(i).onKeyPressed(keyType,keyValue);
            }
        }

    }


}

File: app/src/main/java/com/letianpai/robot/wificonnet/callback/KeyPressCallback.kt
package com.letianpai.robot.wificonnet.callback

/**
 * 按键功能回调
 * @author liujunbin
 */
class KeyPressCallback private constructor() {
    private val keyPressCallbackListenerList = ArrayList<KeyPressCallbackListener?>()

    private object KeyPressCallbackHolder {
        val instance: KeyPressCallback = KeyPressCallback()
    }

    fun setKeyPressCallbackListener(keyPressCallbackListener: KeyPressCallbackListener?) {
        keyPressCallbackListenerList.add(keyPressCallbackListener)
    }

    interface KeyPressCallbackListener {
        fun onKeyPressed(keyType: Int, keyValue: String?)
    }

    fun pressKey(keyType: Int, keyValue: String?) {
        for (i in keyPressCallbackListenerList.indices) {
            if (keyPressCallbackListenerList[i] != null) {
                keyPressCallbackListenerList[i]!!.onKeyPressed(keyType, keyValue)
            }
        }
    }


    companion object {
        const val KEY_TYPE_FUNCTION: Int = 10010
        const val KEY_TYPE_VALUE: Int = 10011

        @JvmStatic
        val instance: KeyPressCallback
            get() = KeyPressCallbackHolder.instance
    }
}
